src/components/Route/CreateRoute.tsx

```diff
@@ -0,0 +1,195 @@
+import { createRoute } from '../../xplat/api';
+import Popup from 'reactjs-popup';
+import 'reactjs-popup/dist/index.css';
+import { User } from '../../xplat/types/user';
+import { RouteType, RouteClassifier, RouteColor, RouteTech } from '../../xplat/types/route';
+import { getAllBoulderClassifiers, getAllTraverseRouteClassifiers, 
+  getAllCompetitionRouteClassifiers,
+  getAllLeadclimbRouteClassifiers,
+  getAllTopropeRouteClassifiers } from '../../xplat/api';
+
+
+const RouteTypeToGetAllClassifiers = (type: RouteType) => {
+  if (type === RouteType.Boulder)
+    return getAllBoulderClassifiers();
+  if (type === RouteType.Traverse)
+    return getAllTraverseRouteClassifiers();
+  if (type === RouteType.Competition)
+    return getAllCompetitionRouteClassifiers();
+  if (type === RouteType.Leadclimb)
+    return getAllLeadclimbRouteClassifiers();
+  return getAllTopropeRouteClassifiers();
+};
+
+const Ropes = [
+  '1', '2', '3', '4', '5', '6', '7', '8', '9'
+];
+
+type refreshRoutesCallback = () => void;
+
+type CreateRouteProps = {
+  refreshRoutes: refreshRoutesCallback;
+  isOpen: boolean;
+  setIsOpen: (arg0: boolean) => void;
+};
+
+// TODO: implement User search component
+const CreateRoute = ({refreshRoutes, isOpen, setIsOpen}: CreateRouteProps) => {
+  const [name, setName] = useState<string>('');
+  const [type, setType] = useState<RouteType>();
+  const [rope, setRope] = useState<number>();
+  const [rawgrade, setRawgrade] = useState<number>();
+  const [description, setDescription] = useState<string>();
+  const [rules, setRules] = useState<RouteTech>();
+  const [routeColor, setRouteColor] = useState<RouteColor>();
+  const [confirmationBoxOpen, setConfirmationBoxOpen] = useState(false);
+  const [thumbnailFile, setThumbnailFile] = useState<File>();
+  const [setter, setSetter] = useState<User>();
+  const [overrideSetterBool, setOverrideSetterBool] = useState<boolean>(true);
+  const [overrideSetterString, setOverrideSetterString] = useState<string>('');
+  const [formError, setFormError] = useState(false);
+
+  function handleFileSelect(event: React.ChangeEvent<HTMLInputElement>) {
+    if (event.target.files === null)
+      return;
+    setThumbnailFile(event.target.files[0]);
+  }
+
+  function checkRouteData(): boolean {
+    if (rawgrade === undefined || name.length === 0 || type === undefined 
+      || rope === undefined || rules === undefined || routeColor === undefined)
+    {
+      setFormError(true);
+      return false;
+    }
+    setFormError(false);
+    return true;
+  }
+
+  async function submitRoute() {
+    // pacify data
+    if (checkRouteData() === false)
+      return;
+    // check that no active route has the same name
+    createRoute(
+      {
+        name: name, 
+        classifier: new RouteClassifier(rawgrade!, type!), 
+        setter: setter,
+        setterRawName: overrideSetterString,
+        description: description,
+        rope: rope,
+        tech: rules,
```

Comment:
this should be using the newest version of xplat with naturalRules field, but this will be resolved once you grab the newest version for RouteColor enum anyway 👍